src/main/java/com/yahoo/memory/BaseBuffer.java

```diff
@@ -88,46 +124,67 @@ public BaseBuffer setPosition(final long position) {
   }
 
   /**
-   * Increments the current position by the given increment
-   * @param increment the given increment
+   * Sets the current position.
+   * Checks that the positional invariants are not violated,
+   * otherwise, throws an {@link IllegalArgumentException}.
+   * @param position the given current position.
    * @return BaseBuffer
    */
-  public BaseBuffer incrementPosition(final long increment) {
-    incrementPosition(pos, increment);
+  public BaseBuffer setPositionAndCheck(final long position) {
+    checkInvariants(start, position, end, capacity);
```

Comment:
Don't you think it should checkValid() too?

Reply:
I thought about it.  Originally, I had no valid checks in BaseBuffer, which only manipulates the local-view positional values.  My feeling was that checking the position invariants was enough, because as soon as the user tries to do any read or write using the new position the validity of the local view will be tested.  

You moved the two valid checks into the package private methods incrementPositionAndCheck/Assert() to streamline the public get/put*Array() methods.  And the public set methods of BaseBuffer use the same package private method for convenience.  
Similarly, you created local checkBounds() and assertBounds() in WritableBufferImpl() to hide the assertValid() and checkValid() calls from the public entry points.  

There is some personal preferences here, but, I think the checkValid methods need only be associated with public methods that actually touch data. Moving them would make it clearer to a developer _why_ they exist where they do.  It has nothing to do with finding stuff with the IDE.

Reply:
Ok.